core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,161 @@
+package ocr2keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"cosmossdk.io/errors"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/keeper_registry_wrapper2_0"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	evm21 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	blockSubscriber    *evm21.BlockSubscriber
+	blockSubChanID     int
+	threadCtrl         utils.ThreadControl
+	lggr               logger.Logger
+	configDigest       [32]byte
+	latestConfigDigest latestConfigDigestGetter
+}
+
+type latestConfigDigestGetter interface {
```

Comment:
you can use config tracker interface here instead of getting it from registry 
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/delegate.go#L961

Reply:
this is the interface https://github.com/smartcontractkit/chainlink/blob/developlang/1.21.1/packages/pkg/mod/github.com/smartcontractkit/chainlink-relay@v0.1.7-0.20231016132219-9c6fdb379595/pkg/types/provider.go#L9